programs/clearing_house/src/controller/orders.rs

```diff
@@ -951,25 +950,26 @@ pub fn fulfill_order_with_amm(
         None
     };
 
-    let (potentially_risk_increasing, _, quote_asset_amount, quote_asset_amount_surplus, pnl) =
-        controller::position::update_position_with_base_asset_amount(
-            base_asset_amount,
-            order_direction,
-            market,
-            user,
-            position_index,
-            mark_price_before,
-            now,
-            maker_limit_price,
-        )?;
-
-    controller::position::update_unsettled_pnl(&mut user.positions[position_index], market, pnl)?;
-
-    let mut unsettled_pnl = pnl;
-
     let (order_post_only, order_ts, order_direction) =
         get_struct_values!(user.orders[order_index], post_only, ts, direction);
 
+    let (
+        potentially_risk_increasing,
+        _,
+        quote_asset_amount,
+        quote_asset_amount_surplus,
+        position_delta,
+    ) = controller::position::calculate_position_delta_with_base_asset_amount(
```

Comment:
we usually use `calculate_` naming convention for math functions where as this mutates the state of the amm, so the name should reflect that